### PR TITLE
Add lookml validation to CI

### PR DESCRIPTION
**Pull request by scholtzan**
Have lookml validation as part of CI. It has happened that faulty lookml was pushed due to looker-hub being out-of-date for the developers. So this should catch these instances. See https://mozilla.slack.com/archives/CKFKC7Y1Y/p1710966870840059

Checklist for reviewer:

When adding a new derived dataset:
- [ ] Ensure that the data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data may be available in [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).
- [ ] Avoid merging a PR that includes the logic of a [core metric](https://docs.telemetry.mozilla.org/metrics/index.html) or complex business logic. The recommendation is to implement core business logic in bigquery-etl. E.g. The [type of search](https://github.com/mozilla/bigquery-etl/blob/a3e59f90326816a2ecaaa3e9d5b57fe9552f7d70/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_daily_v1/query.sql#L781) or the [calculation of DAU or visited URIs](https://github.com/mozilla/bigquery-etl/blob/9bca48821a8a0d40b1700cc14ecd8068d132ed06/sql/moz-fx-data-shared-prod/telemetry_derived/firefox_desktop_exact_mau28_by_dimensions_v1/query.sql).
- [ ] Avoid merging code in Looker Explores/Views that implement analysis with multiple lines of code or that will be likely replicated in the future. Instead, aim for extending an existing dataset to include the required logic, and use [Looker aggregates](https://cloud.google.com/looker/docs/aggregate_awareness) to facilitate the analysis.
- [ ] Avoid merging a PR with logic that requires validation and health checks. It is recommended to implement it in bigquery-etl for full test coverage and failure alerts.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -30,23 +30,48 @@ jobs:
           command: |
             python ./src/sync_dashboards/main.py sync --config lookml_dashboards.yaml
           no_output_timeout: 15m
+  validate-lookml:
+    executor: python/default
+    steps:
+      - checkout
+      - run:
+          name: Install spectacles
+          command: pip install spectacles
```
Comment (kik-kik) [inline .circleci/config.yml:39]: Should we be tagging a specific version of `spectacles`?

**Review by Marlene-M-Hirose — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -30,23 +30,48 @@ jobs:
...
+      - run:
+          no_output_timeout: 20m
+          command: |
+            spectacles lookml \
```
Comment (BenWu) [inline .circleci/config.yml:43]: The docs say this only checks for syntax errors: https://docs.spectacles.dev/cli/tutorials/validators/#the-lookml-validator

Would the issue Teon mentioned be included in this?  I didn't look too deeply but intuitively it sounds like it be caught by the content validator

Comment (scholtzan) [reply .circleci/config.yml:43]: We reproduced Teons issue when we tested it and it did detect it.

Comment (scholtzan) [reply .circleci/config.yml:43]: From what I could tell `spectacles` is just calling the Looker API. And it seems it uses the lookml validation that is also part of the web UI

File: .circleci/config.yml (hunk 1 above)
Comment (scholtzan) [inline .circleci/config.yml:39]:
```suggestion
          command: pip install spectacles==2.4.7
```

File: .circleci/config.yml (hunk 2 above)
Comment (BenWu) [reply .circleci/config.yml:43]: Alright, good enough for me

File: .circleci/config.yml [hunk 3]
```diff
@@ -30,23 +30,48 @@ jobs:
           command: |
             python ./src/sync_dashboards/main.py sync --config lookml_dashboards.yaml
           no_output_timeout: 15m
+  validate-lookml:
+    executor: python/default
+    steps:
+      - checkout
+      - run:
+          name: Install spectacles
+          command: pip install spectacles==2.4.7
```
Comment (scholtzan) [inline .circleci/config.yml:39]:
```suggestion
          command: pip install spectacles==2.3.18
```

**Review by quiiver — APPROVED** [review]

**Comment by scholtzan** [comment]
mmh I'll look into this timeout issue before merging

**Comment by scholtzan** [comment]
This also doesn't seem to be pulling in updates from `looker-hub` :/

**Pull request closed without merging**